### PR TITLE
Assert on specific warnings in tests

### PR DESCRIPTION
**Pull request by taion**
Easier all around just to have one test run that includes backward compat and just asserts what warnings should be there.

### CODE REVIEW

**Comment by taion** [comment]
Easier all around just to have one test run that includes backward compat and just asserts what warnings should be there.

**Comment by timdorr** [comment]
Hmm, why did the CI fail?

**Comment by taion** [comment]
Better question is why it passed locally. Or how this ever worked in the first place. I'm just going to fix the test case.

File: modules/__tests__/transitionHooks-test.js [hunk 1]
```diff
@@ -355,7 +354,7 @@ describe('When a router enters a branch', function () {
     it('calls the onEnter hooks of all routes in that branch', function (done) {
       const newsFeedRouteEnterSpy = spyOn(NewsFeedRoute, 'onEnter').andCallThrough()
       const newsFeedRouteChangeSpy = spyOn(NewsFeedRoute, 'onChange').andCallThrough()
-      const history = useQueries(createHistory)('/inbox')
```
Comment (taion) [inline modules/__tests__/transitionHooks-test.js]: ??!

This was never a valid configuration, either in backward compat or not.

Comment (timdorr) [inline modules/__tests__/transitionHooks-test.js]: Added by #2137

**Pull request merged**